proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -373,14 +383,22 @@ message ActivityInfo {
     // independently-assigned build ID to the database. This case heals automatically once the task is dispatched.
     oneof assigned_build_id {
         // When true, it means this activity is assigned to the build ID of its workflow.
-        google.protobuf.Empty use_workflow_build_id = 35;
+        UseWorkflowBuildId use_workflow_build_id = 35;
```

Comment:
Even if there is a persisted `Empty` value, when reading it the inner fields would be unset. That's fine because:
1) the inner fields only go to the ActivityStarted event, not anywhere else in the server logic
2) the only use of this info in ActivityStarted event is to re-apply redirect rules when rebuilding MS. And that's not applicable before this PR because redirect rule where not supported and applied to any WF to begin with.